dcos_test_utils/dcos_cli.py

```diff
@@ -18,7 +19,40 @@
 
 log = logging.getLogger(__name__)
 
-DCOS_CLI_URL = os.getenv('DCOS_CLI_URL', 'https://downloads.dcos.io/binaries/cli/linux/x86-64/dcos-1.12/dcos')
+
+def _get_dcos_cli_url():
+    """ This returns the DC/OS CLI URL based on the current DC/OS version.
+
+        The DC/OS version is read from gen/calc.py. For DC/OS < 1.14, it will
+        return https://downloads.dcos.io/binaries/cli/linux/x86-64/dcos-1.XX/dcos
+        with the appropriate version. Starting from DC/OS 1.14 it will return the
+        DC/OS version agnostic CLI URL.
+
+        The URL can be overwritten through the DCOS_CLI_URL env var.
+    """
+    download_url = os.getenv('DCOS_CLI_URL')
+    if download_url:
+        return download_url
+
+    try:
+        from gen.calc import DCOS_VERSION
```

Comment:
What if gen.calc is not importable? ( dcos-test-utils does not have a dependency on `gen`) at the moment.

Reply:
This will make dcos-test-utils tied dcos/dcos repo, which is undesirable in the codebase.

Reply:
@orsenthil That appears to leave two options:

* Hardcode a default, allow overrides
* Require the user to set the CLI version

I personally prefer the latter but would like for this PR to move forward.

Reply:
Looking at the code again it seems hard to use dcos-metadata because of the way the class is structured. The CLI is downloaded when the class is instanciated, at this point the cluster URL is not known yet.

I think we should use the latest stable CLI as a default and allow overriding, otherwise we'd have to pretty much change the whole API of the class. WDYT @orsenthil ?

Reply:
Yeah, that sounds better to me too.

* Let's use latest stable CLI as default and allow for overriding. 

Thank you!

Reply:
Great! Looks like we have a way forward with this.

Reply:
Ok, I've updated the PR.